### PR TITLE
extend/os: clean up requires

### PR DESCRIPTION
**Pull request by apainintheneck**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

These need to be after the original module/class definitions for the monkey-patching to work correctly.

### CODE REVIEW

File: Library/Homebrew/extend/os/linux/dev-cmd/update-test.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ module Homebrew
 
   module_function
 
+  alias generic_git_tags git_tags
+
   def git_tags
     tags = generic_git_tags
```
Comment (apainintheneck) [inline Library/Homebrew/extend/os/linux/dev-cmd/update-test.rb:12]: Not sure why it's saying `generic_git_tags` is undefined here.

**Review by MikeMcQuaid — APPROVED** [review]
Good catch, thanks @apainintheneck. 

CC @hyuraku FYI

File: Library/Homebrew/extend/os/linux/dev-cmd/update-test.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/extend/os/linux/dev-cmd/update-test.rb:12]: @apainintheneck I think you need to make the alias in `Library/Homebrew/dev-cmd/update-test.rb` instead, that's what we've done previously here.

Comment (apainintheneck) [reply Library/Homebrew/extend/os/linux/dev-cmd/update-test.rb:12]: That's actually what I tried first and I moved after seeing the same error. I feel like I'm missing something obvious here.

File: Library/Homebrew/extend/os/linux/dev-cmd/update-test.rb [hunk 2]
```diff
@@ -6,6 +6,10 @@ module Homebrew
 
   module_function
 
+  class << self
+    alias generic_git_tags git_tags
+  end
```
Comment (apainintheneck) [inline Library/Homebrew/extend/os/linux/dev-cmd/update-test.rb:11]: This works but is really ugly and hacky. Do we really want to do this or is there a better way to handle `extend/os` stuff with `module_function`?

Comment (MikeMcQuaid) [reply Library/Homebrew/extend/os/linux/dev-cmd/update-test.rb:11]: @apainintheneck That's the best way to hand this with `module_function`. The alternative would be to wrap all methods in the `class << self` and drop `module_function` use entirely.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @apainintheneck!

**Pull request merged**